### PR TITLE
Support sent attachment by url.

### PR DESCRIPTION
**Pull request by appleboy**
Example:

``` go
client.Attachment(To, "image", "https://cdn3.iconfinder.com/data/icons/picons-social/57/16-apple-256.png")
client.Attachment(To, "audio", "https://archive.org/download/testmp3testfile/mpthreetest.mp3")
client.Attachment(To, "video", "http://www.sample-videos.com/video/mp4/480/big_buck_bunny_480p_5mb.mp4")
client.Attachment(To, "file", "http://open.qiniudn.com/where-can-you-use-golang.pdf")
```

screen shot 2016-10-08 at 12 52 19 pm

Signed-off-by: Bo-Yi Wu appleboy.tw@gmail.com

### CODE REVIEW

File: messenger.go [hunk 1]
```diff
@@ -305,6 +305,16 @@ func (m *Messenger) SendWithReplies(to Recipient, message string, replies []Quic
 	return response.TextWithReplies(message, replies)
 }
 
+// Attachment send image, audio, video and file attachment.
```
Comment (paked) [inline messenger.go]: Let's just clear up the grammar here a bit... Maybe something like the following:

> Attachment sends an image, sound, video or a regular file to a given recipient.

File: messenger.go [hunk 2]
```diff
@@ -305,6 +305,16 @@ func (m *Messenger) SendWithReplies(to Recipient, message string, replies []Quic
...
+func (m *Messenger) Attachment(to Recipient, dataType, url string) error {
```
Comment (paked) [inline messenger.go]: When the constant attachment type below is implemented, this signature will need to change too.

File: response.go [hunk 3]
```diff
@@ -130,6 +130,41 @@ func (r *Response) Image(im image.Image) error {
 	return nil
 }
 
+// Attachment send image, audio, video and file attachment
+func (r *Response) Attachment(dataType, url string) error {
```
Comment (paked) [inline response.go]: We should probably get rid of the magic values introduce by the `dataType` string.

Something like the following would be great:

``` go
type AttachmentType string

const (
   ImageAttachment AttachmentType = "image"
   AudioAttachment AttachmentType = "audio"
   VideoAttachment AttachmentType = "video"
   FileAttachment AttachmentType = "file"
)
```

So we can then change this function's signature to `func(to Recipient, dataType AttachmentType, url string) error`

File: response.go [hunk 4]
```diff
@@ -130,6 +130,41 @@ func (r *Response) Image(im image.Image) error {
 	return nil
 }
 
+// Attachment send image, audio, video and file attachment
```
Comment (paked) [inline response.go]: Maybe something like:

> Attachment sends an image, sound, video or a regular file to a chat.

**Review by paked — CHANGES_REQUESTED** [review]
Thanks for this PR! We really need to work on keeping up with Facebook's API changes. I've left a few comments, but everything looks ready to go after they're fixed!

File: response.go (hunk 3 above)
Comment (appleboy) [reply response.go]: I don't know why need defined const value for `AttachmentType`? We can make some example to explain how to use `Attachment` method in the second argument.

File: response.go (hunk 4 above)
Comment (appleboy) [reply response.go]: Done!

File: messenger.go (hunk 1 above)
Comment (appleboy) [reply messenger.go]: Done!

File: response.go (hunk 3 above)
Comment (appleboy) [reply response.go]: We need `AttachmentType` const value unless implement the following method.

``` go
func (m *Messenger) ImageAttachment
func (m *Messenger) AudioAttachment
func (m *Messenger) VideoAttachment
func (m *Messenger) FileAttachment
```

and 

``` diff
m := SendStructuredMessage{
    Recipient: r.to,
    Message: StructuredMessageData{
        Attachment: StructuredMessageAttachment{
-           Type: dataType,
+           Type: FileAttachment,
            Payload: StructuredMessagePayload{
                Url: url,
            },
        },
    },
}
```

Comment (paked) [reply response.go]: I prefer the constant method, it's less repetition.

Comment (appleboy) [reply response.go]: @paked Done. Please review again.

**Comment by appleboy** [comment]
@paked Already Done.

File: messenger.go (hunk 2 above)
Comment (appleboy) [reply messenger.go]: Done!

**Comment by appleboy** [comment]
We can change the example:

``` go
client.Attachment(To, messenger.ImageAttachment, "https://cdn3.iconfinder.com/data/icons/picons-social/57/16-apple-256.png")
client.Attachment(To, messenger.AudioAttachment, "https://archive.org/download/testmp3testfile/mpthreetest.mp3")
client.Attachment(To, messenger.VideoAttachment, "http://www.sample-videos.com/video/mp4/480/big_buck_bunny_480p_5mb.mp4")
client.Attachment(To, messenger.FileAttachment, "http://open.qiniudn.com/where-can-you-use-golang.pdf")
```

**Comment by paked** [comment]
Thanks for the PR yet again @appleboy. It's great to see your continuing support for this project 💯

**Pull request merged**